src/System.Management.Automation/engine/Utils.cs

```diff
@@ -895,113 +895,61 @@ internal static bool IsAdministrator()
 #endif
         }
 
-        internal static bool NativeItemExists(string path)
+        internal static bool ItemExists(string path)
         {
-            bool unusedIsDirectory;
-            Exception unusedException;
+            bool result = false;
+            try
+            {
+                result = ItemExists(path, out bool unused);
+            }
+            catch
+            {
+            }
 
-            return NativeItemExists(path, out unusedIsDirectory, out unusedException);
+            return result;
         }
 
-        // This is done through P/Invoke since File.Exists and Directory.Exists pay 13% performance degradation
-        // through the CAS checks, and are terribly slow for network paths.
```

Comment:
I'm fine with changing to `File.GetAttribute()` because `CAS` is not supported in .NET Core, so the 13% penalty is gone. So I think the biggest concern of using the .NET API is not there anymore.

However, can you check the `"network paths"` scenario?

Reply:
The code come from Windows PowerShell and I don't know why it existed. I could not find any way where we would have throw on "network path not found" and not throw on "local path not found".
If we continue migration from ItemExists to File.Exists/Directory.Exists perhaps we found the amazing case.
Even if it is necessary we should eliminate the exception and replace it with regular check.
Now I suppose it's contrived

Reply:
No, I was saying about the "terribly slow for network paths" part. Is "File.Exists and Directory.Exists" still terribly slow on network paths? I guess not as they all checking for attribute now, just like our implementation here. But better to check out.

Reply:
I see a delay only if I use a non-existent server name. I suppose this usual behavior with DNS name resolving.